code/src/pages/dashboard/dashboard-product/components/dashboard-product-container/DashboardProductContainer.tsx

```diff
@@ -0,0 +1,234 @@
+import { useState } from "react";
+
+import EditIcon from "@mui/icons-material/Edit";
+
+import DashboardTabContainer from "@/layouts/dashboard-layout/components/dashboard-tab-container/DashboardTabContainer";
+
+import PageLoadingFallback from "@/containers/page-loading-fallback/PageLoadingFallback";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import routes from "@/constants/routes";
+import useErrorPageRedirect from "@/hooks/use-error-page-redirect/useErrorPageRedirect";
+import { dashboardProductPageNotFoundErrorConfig } from "@/pages/dashboard/dashboard-product/DashboardProductPage.constants";
+import { DashboarProductContainerProps } from "@/pages/dashboard/dashboard-product/DashboardProductPage.types";
+import { useGetManagerProductQuery } from "@/store/api/productsApi";
+import cn from "@/utils/cn/cn";
+import formatPrice from "@/utils/format-price/formatPrice";
+import getCategoryFromTags from "@/utils/get-category-from-tags/getCategoryFromTags";
+import isErrorWithStatus from "@/utils/is-error-with-status/isErrorWithStatus";
+
+import "@/pages/dashboard/dashboard-product/components/dashboard-product-container/DashboardProductContainer.scss";
+
+const DashboardProductContainer = ({
+  productId
+}: DashboarProductContainerProps) => {
+  const [selectedLocaleIndex, setSelectedLocaleIndex] = useState(0);
+
+  const {
+    data: product,
+    isLoading,
+    error
+  } = useGetManagerProductQuery({ productId });
+
+  const { renderRedirectComponent } = useErrorPageRedirect();
+
+  if (isLoading) {
+    return <PageLoadingFallback className="dashboard-tabs__loading-fallback" />;
+  }
+
+  const shouldShowNotFoundError =
+    !product && isErrorWithStatus(error) && error.status === 404;
+
+  if (shouldShowNotFoundError) {
+    return renderRedirectComponent(dashboardProductPageNotFoundErrorConfig);
+  }
+
+  if (!product || error) {
+    return <AppTypography translationKey="errors.somethingWentWrong" />;
+  }
+
+  const languageButtons = product.productTranslations.map(
+    ({ languageCode }, index) => {
+      const buttonClassNames = cn(
+        "dashboard-product__language-button",
+        index === selectedLocaleIndex &&
+          "dashboard-product__language-button--active"
+      );
+
+      const handleClick = () => setSelectedLocaleIndex(index);
+
+      return (
+        <AppButton
+          variant="contained"
+          size="small"
+          key={languageCode}
+          className={buttonClassNames}
+          data-testid="product-language-button"
+          onClick={handleClick}
+        >
+          <AppTypography translationKey={`language.full.${languageCode}`} />
+        </AppButton>
+      );
+    }
+  );
+
+  const productDescription = product.productTranslations[
+    selectedLocaleIndex
+  ].description
+    .split("/")
+    .join("\n");
```

Comment:
I suggest creating a util `parseProductDescription`, because we have similar logic in `ProductDetailsContainer`